src/util/color/predefinedcolorpalettes.cpp

```diff
@@ -241,367 +241,333 @@ constexpr mixxx::RgbColor kTritKeyColor10(0x666600);
-
-const ColorPalette PredefinedColorPalettes::kDefaultTrackColorPalette =
-        mixxx::PredefinedColorPalettes::kMixxxTrackColorPalette;
-
-const ColorPalette PredefinedColorPalettes::kDefaultKeyColorPalette =
-        mixxx::PredefinedColorPalettes::kMixxxKeyColorPalette;
-
-const QList<ColorPalette> PredefinedColorPalettes::kPalettes{
-        // Hotcue Color Palettes
-        mixxx::PredefinedColorPalettes::kMixxxHotcueColorPalette,
-        mixxx::PredefinedColorPalettes::kSeratoDJProHotcueColorPalette,
-        mixxx::PredefinedColorPalettes::kRekordboxCOLD1HotcueColorPalette,
-        mixxx::PredefinedColorPalettes::kRekordboxCOLD2HotcueColorPalette,
-        mixxx::PredefinedColorPalettes::kRekordboxCOLORFULHotcueColorPalette,
-        // Track Color Palettes
-        mixxx::PredefinedColorPalettes::kMixxxTrackColorPalette,
-        mixxx::PredefinedColorPalettes::kRekordboxTrackColorPalette,
-        mixxx::PredefinedColorPalettes::kSeratoDJProTrackColorPalette,
-        mixxx::PredefinedColorPalettes::kTraktorProTrackColorPalette,
-        mixxx::PredefinedColorPalettes::kVirtualDJTrackColorPalette,
-        // Key Color Palettes
-        mixxx::PredefinedColorPalettes::kMixxxKeyColorPalette,
-        mixxx::PredefinedColorPalettes::kTraktorKeyColorPalette,
-        mixxx::PredefinedColorPalettes::kMIKKeyColorPalette,
-        mixxx::PredefinedColorPalettes::kProtKeyColorPalette,
-        mixxx::PredefinedColorPalettes::kDeutKeyColorPalette,
-        mixxx::PredefinedColorPalettes::kTritKeyColorPalette,
-};
-
-const mixxx::RgbColor PredefinedColorPalettes::kDefaultCueColor =
-        kSchemaMigrationReplacementColor;
-
-const mixxx::RgbColor PredefinedColorPalettes::kDefaultLoopColor =
-        kColorMixxxWhite;
-
+namespace predefinedcolorpalettes {
+
+const PredefinedColorPalettes& get() {
+    // All Rekordbox Palette types, share the same color, but their default colors
+    // are not in the selection offered to the user. This usecase is not supported
+    // by mixxx's ColorPalette. The compromise is slightly altering the default
+    // colors used so they're part of the selection. The difference between these
+    // colors should be imperceptible for the unknowing user.
+    const QList<mixxx::RgbColor> kRekordboxColorsSelection = {
+            kRekordboxHotcueColor1,
+            kRekordboxHotcueColor2,
+            kRekordboxHotcueColor3,
+            kRekordboxHotcueColor4,
+            kRekordboxHotcueColor5,
+            kRekordboxHotcueColor6,
+            kRekordboxHotcueColor7,
+            kRekordboxHotcueColor8,
+            kRekordboxHotcueColor9,
+            kRekordboxHotcueColor10,
+            kRekordboxHotcueColor11,
+            kRekordboxHotcueColor12,
+            kRekordboxHotcueColor13,
+            kRekordboxHotcueColor14,
+            kRekordboxHotcueColor15,
+            kRekordboxHotcueColor16,
+    };
+
+    const static PredefinedColorPalettes kPalettes{
```

Comment:
Returning a local static is one way ti implement a singleton pattern. How about fully make use of that here? 
as in a `PredefinedColorPalettes::instance()` function.